Zend/Optimizer/zend_inference.c

```diff
@@ -3371,9 +3371,19 @@ static zend_always_inline zend_result _zend_update_type_info(
 				UPDATE_SSA_OBJ_TYPE(ce, 0, ssa_op->result_def);
 			} else if ((t1 & MAY_BE_CLASS) && ssa_op->op1_use >= 0 && ssa_var_info[ssa_op->op1_use].ce) {
 				UPDATE_SSA_OBJ_TYPE(ssa_var_info[ssa_op->op1_use].ce, ssa_var_info[ssa_op->op1_use].is_instanceof, ssa_op->result_def);
+				if (!ssa_var_info[ssa_op->result_def].is_instanceof) {
+					ce = ssa_var_info[ssa_op->op1_use].ce;
+				}
 			} else {
 				UPDATE_SSA_OBJ_TYPE(NULL, 0, ssa_op->result_def);
 			}
+			/* New objects without constructors cannot escape. */
+			if (ce
+			 && !ce->constructor
+			 && !ce->create_object
+			 && ce->default_object_handlers->get_constructor == zend_std_get_constructor) {
```

Comment:
The last line is not necessary. If ``create_object`` is NULL then ``ce->default_object_handlers->get_constructor == zend_std_get_constructor`` is true.

Reply:
As discussed in the other issue, the default handlers may be changed without specifying `create_object`. `create_object` may still be used to swap the handlers. Support for this could be dropped in the future, essentially making `zend_object->handlers` and implementation detail, i.e. a cached pointer to `object->ce->default_object_handlers` to avoid a pointer indirection.